### DISCUSSION TITLE
Unit Tests

### DISCUSSION THREAD

**Discussion by MixusMinimax** [android/ndk-samples#1082] [Q&A] [answered]
Hey, I'm trying to add unit tests for my C++ code.

I have looked at the unit-test example in this repo, and have some questions:

The tests are running on my phone, so all the test targets are always compiled, and included in the final APK, just that these libraries aren't loaded outside of the kotlin unit tests.

The example gradle build excludes these `.so` files from the build, because you don't want them in the apk, which makes sense for the final release of an application. However, then the tests don't work, as they need those libraries.

Are you somehow running the native tests on the host machine, and therefore don't need to install them onto the phone?

Also, are unit tests for ndk just not something we are supposed to do right now? I could also just compile parts of my C++ code that doesn't use anything android related (utility classes etc.) and test them using normal ctest practices.

**Comment by DanAlbert** [comment, marked as answer]
This was https://issuetracker.google.com/236913987. We're on a new enough version of AGP now to have the fix, but https://github.com/android/ndk-samples/blob/9d38db97bd23f5a3c7ea2a3e711402e4a9f78551/unit-test/app/build.gradle#L31 needs to be updated to use it. https://github.com/android/ndk-samples/issues/1083
